src/runtime_src/hip/api/hip_stream.cpp

```diff
@@ -2,35 +2,79 @@
 // Copyright (C) 2023-2024 Advanced Micro Device, Inc. All rights reserved.
 
 #include "hip/core/common.h"
+#include "hip/core/event.h"
 #include "hip/core/stream.h"
 
 namespace xrt::core::hip {
-static hipStream_t
+// In Hip, based on flags we can create default or non blocking streams.
+// If application doesn't explicitly specify stream we use default stream
+// for such operations. The default stream also has two modes legacy or per thread.
+// Legacy default stream is also called as NULL stream. Null stream waits on all explictly
+// created default streams in the same context when an operation is enqueued and explicitly
+// created default streams wait on null stream in that context.
+// Per thread stream is also default stream but is created per thread per context and waits
+// on null stream of that context.
+
```

Comment:
This comment is convoluted. Can you create an ASCII diagram instead and label them with legacy default stream, per thread default stream and who is waiting on whom?

Reply:
sure @sonals I will add it. Thanks